tf_encrypted/convert/convert_test.py

```diff
@@ -887,8 +889,17 @@ def export(x: tf.Tensor, filename: str, sess=None):
     should_close = True
     sess = tf.Session()
 
-  pred_node_names = ["output"]
-  tf.identity(x, name=pred_node_names[0])
+  last_node_name = sess.graph.as_graph_def().node[-1].name
+
+  # if the op returns a list of tensors, we want avoid
+  # adding a tf.identiy op so the the list of tensors doesn't
+  # get packed.
+  if last_node_name != "output":
+    pred_node_names = ["output"]
+    tf.identity(x, name=pred_node_names[0])
```

Comment:
hm, would it be easier if we just check to see if x is a list instead of a single tensor, and then do `out = [tf.identity(xi) for xi in x]`?

Reply:
perhaps I'm misunderstanding what the problem is here though

Reply:
Just changed  the approach a bit. However I am getting an error for special ops: 'required_space_to_batch_paddings':

```
def _register_specop(self, node, specop_scope_dict):
      """Handle special op registration."""
      input_list = specop_scope_dict['inputs']
      output_list = specop_scope_dict['outputs']
    
      # Handle edge cases if the ops return multiple outputs
      op_handler = self.registry[specop_scope_dict['op']]
    
      nodes = specop_scope_dict['interiors']
      if not nodes:
        nodes = node
      outs = op_handler(self, nodes, input_list)
      if isinstance(outs, (list, tuple)):
        for i, x in enumerate(outs):
>         self.outputs[output_list[i]] = x
E         IndexError: list index out of range

tf_encrypted/convert/convert.py:147: IndexError
```

`tf.identity` was also probably packing the outputs from `required_space_to_batch_paddings` that's why probably we haven't seen this bug :)

Reply:
Ok... so the problem is not the converter, it's the test. `required_space_to_batch_paddings` output two nodes in the graph: padding and crop. Whereas tf.split is always the last node. I tried in `graph_util.convert_variables_to_constants` to list two nodes as output but it doesn't work properly when the graph gets exported. 

I see two solution for now:
- we keep the original test with `tf.identity` it's just that the output for `required_space_to_batch_paddings`  and `split` will get packed. or
- we use the second version of the test where I renamed the node split as output. In this case the output for split won't be packed but, we will leave the output for `required_space_to_batch_paddings`. but the code looks less clean.